src/test/key_tests.cpp

```diff
@@ -150,6 +150,14 @@ BOOST_AUTO_TEST_CASE(key_test1)
     BOOST_CHECK(key2C.SignCompact(hashMsg, detsigc));
     BOOST_CHECK(detsig == ParseHex("1c52d8a32079c11e79db95af63bb9600c5b04f21a9ca33dc129c2bfa8ac9dc1cd561d8ae5e0f6c1a16bde3719c64c2fd70e404b6428ab9a69566962e8771b5944d"));
     BOOST_CHECK(detsigc == ParseHex("2052d8a32079c11e79db95af63bb9600c5b04f21a9ca33dc129c2bfa8ac9dc1cd561d8ae5e0f6c1a16bde3719c64c2fd70e404b6428ab9a69566962e8771b5944d"));
+
+    // test EC Diffie-Hellman secret computation
+    CPrivKey secret1;
+    CPrivKey secret2;
+    BOOST_CHECK(key1.ComputeECDHSecret(pubkey2, secret1));
+    BOOST_CHECK(key2.ComputeECDHSecret(pubkey1, secret2));
+    BOOST_CHECK(secret1.size() == 32);
+    BOOST_CHECK(secret1 == secret2);
 }
```

Comment:
Needs a test where the pubkey doesn't decode.

Reply:
Added

Reply:
Thanks but please also add one that could actually have come in over the wire for BIP151, -- the first byte isn't sent there.  What I'm hoping we check for is that it correctly rejects points that aren't on the curve here, rather than performing operations on the twist.

Reply:
The new `pubkeydata[9] = 0xFF;` line checks that, right? Maybe add a comment to clarify?